src/network/networkd-dhcp4.c

```diff
@@ -81,6 +83,12 @@ static int link_set_dhcp_routes(Link *link) {
         if (!link->network->dhcp_use_routes)
                 return 0;
 
+        /* When the interface is part of an VRF use the VRFs routing table, unless
+         * there is a another table specified. */
+        table = link->network->dhcp_route_table;
+        if (table == RT_TABLE_MAIN && link->network->vrf != NULL)
+                table = VRF(link->network->vrf)->table;
```

Comment:
hmm, but shouldn't the user be able to explicitly set the table to RT_TABLE_MAIN if he likes to? with this change this wouldn't be possible, would it?

maybe maintain a boolean link_network->dhcp_route_table_set or so, which is turned on if the user actually uses RouteTable= in the [DHCP] section. And then, when it is unset we add an automatic logic here and use the VRF one if that applies and RT_TABLE_MAIN if it doesn't, if you follow what I mean?

That way, if the user sets something it takes effect, and if he doesn't things happen automatically the right way

Reply:
also, this probably deserves a brief explanation in the man page

Reply:
also, wouldn#t the same apply to ipv6 RA, which has similar facilities?

Reply:
I'll handle all of the points above. They are very valid points.

The IPv6 case is not relevant in the SLAAC case. The IPv6 case has to be handled similary.

I'll update the PR shortly.